MeshGeoToolsLib/AppendLinesAlongPolyline.cpp

```diff
@@ -32,9 +32,18 @@ MeshLib::Mesh* appendLinesAlongPolylines(const MeshLib::Mesh &mesh, const GeoLib
 	// copy existing nodes and elements
 	std::vector<MeshLib::Node*> vec_new_nodes = MeshLib::copyNodeVector(mesh.getNodes());
 	std::vector<MeshLib::Element*> vec_new_eles = MeshLib::copyElementVector(mesh.getElements(), vec_new_nodes);
-	unsigned max_matID = 0;
-	for (auto e : vec_new_eles)
-		max_matID = std::max(max_matID, e->getValue());
+
+	std::vector<int> new_mat_ids;
+	{
+		auto mat_ids = mesh.getProperties().getPropertyVector<int>("MaterialIDs");
+		if (mat_ids) {
+			new_mat_ids.resize((*mat_ids).size());
```

Comment:
reserve + push_back is faster, because no initialization happens as with resize.